contracts/utils/contracts/src/LibRichErrors.sol

```diff
@@ -0,0 +1,58 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+
+library LibRichErrors {
```

Comment:
Why add this, rather than replacing `RichErrors`?

Reply:
I had to add this so that I could support "library-style inheritance" (not a technical term) via the "using ... for ..." syntax. I tried to increase code reuse by using LibRichErrors to implement all of the functions in RichErrors, but it didn't seem worth it because the functions were the same size.

Reply:
I wonder if we should just have `RichErrors.sol` be an inheritable thin wrapper for this library.